esp-hal/src/lib.rs

```diff
@@ -570,61 +570,6 @@ mod critical_section_impl {
     }
 }
 
-/// FlashSafeDma
-///
-/// The embedded-hal traits make no guarantees about
-/// where the buffers are placed. The DMA implementation in Espressif chips has
-/// a limitation in that it can only access the RAM address space, meaning data
-/// to be transmitted from the flash address space must be copied into RAM
-/// first.
-///
-/// This wrapper struct should be used when a peripheral using the DMA engine
-/// needs to transmit data from flash (ROM) via the embedded-hal traits. This is
-/// often a `const` variable.
-///
-/// Example usage using [`spi::master::dma::SpiDma`]
-/// ```rust, ignore
-/// const ARRAY_IN_FLASH = [0xAA; 128];
-///
-/// let spi = SpiDma::new(/* */);
-///
-/// spi.write(&ARRAY_IN_FLASH[..]).unwrap(); // error when transmission starts
-///
-/// let spi = FlashSafeDma::new(spi);
-///
-/// spi.write(&ARRAY_IN_FLASH[..]).unwrap(); // success
-/// ```
-pub struct FlashSafeDma<T, const SIZE: usize> {
```

Comment:
Removing this needs to be mentioned in the changelog

Reply:
Ah yeah fair enough. So two entries in the changelog file with the same PR number? One in Add and one in Remove?

Reply:
Sure, why not, there's already one entry in Add and one in Change :) Just a mention so that the API doesn't disappear without a trace.